guides/security.md

```diff
@@ -38,3 +42,39 @@ store them in environment variables, then access them in your code from `process
 navigate to your environment, click _EDIT_, then under _Environment Variables_, click _ADD VARIABLE_.
 
 ![networking](/images/security/environment-variables.png)
+
+## Putting your site behind basic auth
+
+If you would like to protect your site behind [HTTP Basic authentication](https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication) schema, you can respond with a challenge per [environment](environments) (note reading user and password from the environment):
+
+```js
+const router = new Router()
+
+// Use HTTP Basic authentication if authentication user and password are defined in the environment.
+if (process.env.BASIC_AUTH_USER && process.env.BASIC_AUTH_PASSWORD) {
+  const BASIC_AUTH_TOKEN = Buffer.from(`${process.env.BASIC_AUTH_USER}:${process.env.BASIC_AUTH_PASSWORD}`).toString('base64')
+
+  router
+    // Provide challenge when the Authorization header is not present.
+    .match({
+      path: '/:path*',
+      headers: {
+        Authorization: null, // null matching is true when the given header is not present.
+      }
+    }, ({ send, setResponseHeader }) => {
+      setResponseHeader('WWW-Authenticate', 'Basic realm="Access to the site"')
+      send('', 401, 'Unauthorized')
+    })
+    // Block the request when the Authorization header is not matching the basic auth token.
+    .match({
+      path: '/:path*',
+      headers: {
+        Authorization: new RegExp(`^Basic ((?!${BASIC_AUTH_TOKEN}).)*$`)
+      }
+    }, ({ send }) => {
+      send('', 403, 'Forbidden')
+    })
```

Comment:
```suggestion
new Router()
  .requireBasicAuth({ 
    username: process.env.BASIC_AUTH_USERNAME, 
    password: process.env.BASIC_AUTH_PASSWORD 
  })
  // ... the rest of your router...
```

The example above allows you to configure basic auth on preproduction environments by adding `BASIC_AUTH_USERNAME` and `BASIC_AUTH_PASSWORD` environment variables, but will not enforce basic auth on your production environment as long as those variables are not present there.